### PR TITLE
add recipe for marshmallow-dataclass

### PR DESCRIPTION
**Pull request by ElenaBossolini**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml"
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x ] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [x ] If static libraries are linked in, the license of the static library is packaged.
- [x ] Build number is 0
- [x ] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x ] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by lovasoa** [comment]
> [x ] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

I am willing to

>  remove version constraint on python requirement 

The package is not compatible with python < 3.6 , though.

**Comment by ElenaBossolini** [comment]
Hi @conda-forge/staged-recipes, kindly review this recipe PR. Can you please also tackle to @lovasoa 's comment 

> The package is not compatible with python < 3.6 , though.

Do we need to change the recipe in some way?

File: recipes/marshmallow-dataclass/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  name: "{{ conda_name }}"
+  version: "{{ version }}"
+
+source:
+  url: "https://pypi.io/packages/source/{{ pypi_name[0] }}/{{ pypi_name }}/{{ pypi_name }}-{{ version }}.tar.gz"
+  sha256: "cde644b272c2284f1212f0e0d25bdb9eaf7d58f1423ae82b712ba43db229d618"
+
+build:
+  number: 0
+  script: "{{ PYTHON }} -m pip install . -vv"
```
Comment (synapticarbors) [inline recipes/marshmallow-dataclass/meta.yaml:15]: This recipe looks like it should probably be `noarch:python`:

https://conda-forge.org/docs/maintainer/knowledge_base.html#noarch-python

File: recipes/marshmallow-dataclass/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+requirements:
+  host:
```
Comment (synapticarbors) [inline recipes/marshmallow-dataclass/meta.yaml:18]: `host` likely only needs `python` and `pip`. The rest are only runtime dependencies.

File: recipes/marshmallow-dataclass/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    - marshmallow-enum
+    - pip
+    - python
+    - typeguard
+    - typing_inspect
+  run:
+    - dataclasses
+    - marshmallow >=3.0.0,<4.0
+    - marshmallow-enum
+    - python
```
Comment (synapticarbors) [inline recipes/marshmallow-dataclass/meta.yaml:30]: change `python` here to `python >=3.6`. Do the same for the `python` under `host`.

File: recipes/marshmallow-dataclass/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+  license: MIT
+  license_family: MIT
+  license_file: LICENSE
+  summary: "Python library to convert dataclasses into marshmallow schemas."
+  doc_url: "https://lovasoa.github.io/marshmallow_dataclass/html/index.html"
+  dev_url: "https://github.com/lovasoa/marshmallow_dataclass"
+
+extra:
+  recipe-maintainers:
+    - lovasoa
```
Comment (scopatz) [inline recipes/marshmallow-dataclass/meta.yaml:49]: @lovasoa - are you willing to be a maintainer?

Comment (lovasoa) [reply recipes/marshmallow-dataclass/meta.yaml:49]: Yes I do !

https://github.com/conda-forge/staged-recipes/pull/12730#issuecomment-698955408

File: recipes/marshmallow-dataclass/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+
+requirements:
+  host:
+    - dataclasses
+    - marshmallow >=3.0.0,<4.0
+    - marshmallow-enum
+    - pip
+    - python
+    - typeguard
+    - typing_inspect
```
Comment (lovasoa) [inline recipes/marshmallow-dataclass/meta.yaml:25]:
```suggestion
    - pip
    - python >=3.6
```

File: recipes/marshmallow-dataclass/meta.yaml (hunk 3 above)
Comment (lovasoa) [inline recipes/marshmallow-dataclass/meta.yaml:30]:
```suggestion
    - python >=3.6
```

File: recipes/marshmallow-dataclass/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+package:
+  name: "{{ conda_name }}"
+  version: "{{ version }}"
+
+source:
+  url: "https://pypi.io/packages/source/{{ pypi_name[0] }}/{{ pypi_name }}/{{ pypi_name }}-{{ version }}.tar.gz"
+  sha256: "cde644b272c2284f1212f0e0d25bdb9eaf7d58f1423ae82b712ba43db229d618"
+
+build:
+  number: 0
```
Comment (lovasoa) [inline recipes/marshmallow-dataclass/meta.yaml:14]:
```suggestion
  noarch: python
  number: 0
```

**Review by lovasoa — CHANGES_REQUESTED** [review]

**Comment by ElenaBossolini** [comment]
@synapticarbors , thank you for the review! One question: the first time I have committed the file, I only had python constraints (`>=3.6`) on the `run` and the conda-forge-linter did complain that this requirement should have not been there. On the other hand, having the constraint both on the `host` and on the `run` makes the conda-forge-linter say that the recipes look excellent. Why is this?

**Comment by synapticarbors** [comment]
@ElenaBossolini -- Are you referring to the message from the linter that said, "Non noarch packages should have python requirement without any version constraints."? Basically if you have a non-noarch package, then the system wants the CI matrix to decide what versions of python to build for and anything you want to skip should use a `skip: true  # [py<36]` type entry. For `noarch: python`, then we want the constraint to be under the dependencies sections.

**Comment by ElenaBossolini** [comment]
Thanks for the explanation, @synapticarbors

**Pull request merged**